apps/designer/app/designer/features/style-panel/controls/toggle/toggle-control.tsx

```diff
@@ -0,0 +1,66 @@
+import {
+  styled,
+  Flex,
+  Tooltip,
+  ToggleGroupRoot,
+  ToggleGroupItem,
+} from "@webstudio-is/design-system";
+import { StyleProperty } from "@webstudio-is/react-sdk";
+import { useIsFromCurrentBreakpoint } from "../../shared/use-is-from-current-breakpoint";
+
+export type ToggleGroupControlProps = {
+  property: StyleProperty | StyleProperty[];
+  value: string;
+  items: { child: JSX.Element; label: string; value: string }[];
+  onValueChange?: (value: string) => void;
+};
+
+export const ToggleGroupControl = ({
+  property,
+  value = "",
+  items = [],
+  onValueChange,
+}: ToggleGroupControlProps) => {
+  const isCurrentBreakpoint = useIsFromCurrentBreakpoint(property);
+  return (
+    <ToggleGroupRoot
+      type="single"
+      value={value}
+      onValueChange={onValueChange}
+      css={{ width: "fit-content" }}
+    >
+      {items.map(({ child, label, value }, index) => {
+        return (
+          <ToggleGroupControlItem
+            key={index}
+            value={value}
+            state={isCurrentBreakpoint ? "set" : undefined}
+          >
+            <Tooltip content={label} delayDuration={0}>
+              <Flex>{child}</Flex>
+            </Tooltip>
+          </ToggleGroupControlItem>
+        );
+      })}
+    </ToggleGroupRoot>
+  );
+};
+
+const ToggleGroupControlItem = styled(ToggleGroupItem, {
+  variants: {
+    state: {
+      set: {
+        "&[data-state=on]": {
```

Comment:
why do you need to override the on state this way? is that fixing a specificity issue?

Reply:
damn, this isn't great from stitches side ... what options do we have?